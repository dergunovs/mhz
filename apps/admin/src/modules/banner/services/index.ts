import { ComputedRef, Ref } from 'vue';

import { api, useQuery, useMutation, IPageQuery, convertParams } from 'mhz-helpers';
import { API_BANNER, IBanner, IBaseReply, IFilterData } from 'mhz-contracts';

export function getBanners(query: Ref<IPageQuery | number>) {
  async function fn() {
    const params = convertParams(query);

    const { data } = await api.get<{ data: IBanner[]; total: number; filters: IFilterData }>(API_BANNER, { params });

    return data;
  }

  return useQuery({ queryKey: [API_BANNER, query], queryFn: fn });
}

export function getBanner(id?: ComputedRef<string | string[]>) {
  async function fn() {
    if (!id?.value) return null;

    const { data } = await api.get<{ data: IBanner }>(`${API_BANNER}/${id.value}`);

    return data.data;
  }

  return useQuery({ queryKey: [API_BANNER, id], queryFn: fn });
}

export function postBanner(options: object) {
  async function fn(formData: IBanner) {
    const { data } = await api.post<IBaseReply>(API_BANNER, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}

export function updateBanner(options: object) {
  async function fn(formData: IBanner) {
    const { data } = await api.patch<IBaseReply>(`${API_BANNER}/${formData._id}`, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}

export function deleteBanner(options: object) {
  async function fn(id?: string) {
    if (!id) return null;

    const { data } = await api.delete<IBaseReply>(`${API_BANNER}/${id}`);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}
