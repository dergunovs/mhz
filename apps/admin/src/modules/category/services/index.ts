import { ComputedRef } from 'vue';

import { api, useQuery, useMutation } from 'mhz-helpers';
import { API_CATEGORY, ICategory, IBaseReply } from 'mhz-contracts';

export function getCategories(options?: object) {
  async function fn() {
    const { data } = await api.get<{ data: ICategory[] }>(API_CATEGORY);

    return data.data;
  }

  return useQuery({ queryKey: [API_CATEGORY], queryFn: fn, ...options });
}

export function getCategory(id?: ComputedRef<string | string[]>) {
  async function fn() {
    if (!id?.value) return null;

    const { data } = await api.get<{ data: ICategory }>(`${API_CATEGORY}/${id.value}`);

    return data.data;
  }

  return useQuery({ queryKey: [API_CATEGORY, id], queryFn: fn });
}

export function postCategory(options: object) {
  async function fn(formData: ICategory) {
    const { data } = await api.post<IBaseReply>(API_CATEGORY, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}

export function updateCategory(options: object) {
  async function fn(formData: ICategory) {
    const { data } = await api.patch<IBaseReply>(`${API_CATEGORY}/${formData._id}`, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}

export function deleteCategory(options: object) {
  async function fn(id?: string) {
    if (!id) return null;

    const { data } = await api.delete<IBaseReply>(`${API_CATEGORY}/${id}`);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}
