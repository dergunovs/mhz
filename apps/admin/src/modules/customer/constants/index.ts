export const URL_CUSTOMER = '/customers';

export const API_CUSTOMER = 'customer';
