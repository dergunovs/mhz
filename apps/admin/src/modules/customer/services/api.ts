import { Ref, ComputedRef } from 'vue';

import { api, IPageQuery, convertParams } from 'mhz-helpers';
import { API_CUSTOMER, ICustomer } from 'mhz-contracts';

export async function getCustomersApi(query: Ref<IPageQuery | number>) {
  const params = convertParams(query);

  const { data } = await api.get<{ data: ICustomer[]; total: number }>(API_CUSTOMER, { params });

  return data;
}

export async function getCustomerApi(id?: ComputedRef<string | string[]>) {
  if (!id?.value) return null;

  const { data } = await api.get<{ data: ICustomer }>(`${API_CUSTOMER}/${id.value}`);

  return data.data;
}
