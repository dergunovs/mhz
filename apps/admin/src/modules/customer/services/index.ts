import { Ref, ComputedRef } from 'vue';

import { api, useQuery, IPageQuery, convertParams } from 'mhz-helpers';
import { API_CUSTOMER, ICustomer } from 'mhz-contracts';

export function getCustomers(query: Ref<IPageQuery | number>) {
  async function fn() {
    const params = convertParams(query);

    const { data } = await api.get<{ data: ICustomer[]; total: number }>(API_CUSTOMER, { params });

    return data;
  }

  return useQuery({ queryKey: [API_CUSTOMER, query], queryFn: fn });
}

export function getCustomer(id?: ComputedRef<string | string[]>) {
  async function fn() {
    if (!id?.value) return null;

    const { data } = await api.get<{ data: ICustomer }>(`${API_CUSTOMER}/${id.value}`);

    return data.data;
  }

  return useQuery({ queryKey: [API_CUSTOMER, id], queryFn: fn });
}
