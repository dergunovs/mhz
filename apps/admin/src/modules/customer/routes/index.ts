import { URL_CUSTOMER } from '@/customer/constants';

export const customerRoutes = [
  {
    path: URL_CUSTOMER,
    name: 'CustomerList',
    component: () => import('@/customer/pages/CustomerListPage.vue'),
  },
  {
    path: `${URL_CUSTOMER}/:id`,
    name: 'Customer',
    component: () => import('@/customer/pages/CustomerPage.vue'),
  },
];
