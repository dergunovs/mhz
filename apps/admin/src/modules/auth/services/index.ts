import { Ref } from 'vue';
import { useMutation, useQuery } from '@tanstack/vue-query';

import { ILoginFormData } from '@/auth/interface';
import { API_GET_CHECK_AUTH, API_POST_LOGIN } from '@/auth/constants';

import { api } from '@/common/services/api';

export function postLogin(formData: Ref<ILoginFormData>, options: object) {
  async function fn(): Promise<string | undefined> {
    const { data } = await api.post(API_POST_LOGIN, formData.value);

    return data.token;
  }

  return useMutation({
    mutationKey: [API_POST_LOGIN, formData],
    mutationFn: fn,
    ...options,
  });
}

export function getCheckAuth(options: object) {
  async function fn() {
    const { data } = await api.get(API_GET_CHECK_AUTH);

    return data;
  }

  return useQuery({
    queryKey: [API_GET_CHECK_AUTH],
    queryFn: fn,
    ...options,
  });
}
