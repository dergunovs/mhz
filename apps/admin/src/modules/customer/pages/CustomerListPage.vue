<template>
  <div>
    <PageTitle :links="links">{{ title }}</PageTitle>

    <div :class="$style.page">
      <CustomerList
        :customers="customers"
        v-model="query.sort"
        @reset="(value: ISortOption) => resetQuery(value)"
        data-test="customer-list-page-list"
      />

      <UiPagination
        v-show="customers?.length"
        :page="query.page"
        :total="total"
        @update="(value: number) => setQueryPage(setPage(value, query.page))"
      />
    </div>
  </div>
</template>

<script setup lang="ts">
import { useHead } from '@unhead/vue';

import { UiPagination } from 'mhz-ui';
import { usePagination, usePage, ISortOption } from 'mhz-helpers';

import PageTitle from '@/layout/components/PageTitle.vue';
import CustomerList from '@/customer/components/CustomerList.vue';

import { getCustomers } from '@/customer/services';
import { URL_CUSTOMER } from '@/customer/constants';
import { URL_MAIN } from '@/common/constants';

const { query, resetQuery, setQueryPage } = usePage();

const { data } = getCustomers(query);

const { data: customers, total, setPage } = usePagination(data);

const title = 'Customers';

const links = [
  { url: URL_MAIN, title: 'Main' },
  { url: URL_CUSTOMER, title },
];

useHead({
  title,
});
</script>

<style module lang="scss">
.page {
  display: flex;
  flex-direction: column;
  gap: 32px;
}
</style>
