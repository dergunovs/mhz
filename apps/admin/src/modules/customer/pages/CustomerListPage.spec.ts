import { DefineComponent } from 'vue';
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { ICustomer } from 'mhz-contracts';
import { dataTest } from 'mhz-helpers';

import CustomerListPage from './CustomerListPage.vue';

import { mockQueryReply, wrapperFactory } from '@/common/test';
import * as customerServices from '@/customer/services';
import { CUSTOMERS } from '@/customer/fixtures';

const spyGetCustomers = vi.spyOn(customerServices, 'getCustomers').mockReturnValue(mockQueryReply(CUSTOMERS));

const customerList = dataTest('customer-list-page-list');

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(CustomerListPage, {});
});

enableAutoUnmount(afterEach);

describe('CustomerListPage', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(CustomerListPage)).toBeTruthy();
  });

  it('gets customers and passes them to list props', async () => {
    expect(spyGetCustomers).toBeCalledTimes(1);

    expect(
      wrapper.findComponent<DefineComponent<{ customers: ICustomer[] }>>(customerList).vm.$props.customers
    ).toEqual(CUSTOMERS.data);
  });
});
