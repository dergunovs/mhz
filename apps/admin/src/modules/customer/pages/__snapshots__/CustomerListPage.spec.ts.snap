// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomerListPage > matches snapshot 1`] = `
"<div>
  <page-title-stub links="[object Object],[object Object]"></page-title-stub>
  <div class="_page_98c3f0">
    <customer-list-stub customers="[object Object],[object Object]" modelvalue="[object Object]" data-test="customer-list-page-list"></customer-list-stub>
    <ui-pagination-stub page="1" total="1"></ui-pagination-stub>
  </div>
</div>"
`;
